### PR TITLE
chore(comments): fix some typos and unify comments

### PR DESCRIPTION
**Pull request by rettinghaus**
This PR fixes some minor typos I stumbled upon and unifies most of the `TODO` comments to be highlighted. 

### Legal Boilerplate

Look, I get it. The entity doing business as "Sentry" was incorporated in the State of Delaware in 2015 as Functional Software, Inc. and is gonna need some rights from me in order to utilize my contributions in this here PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Sentry can use, modify, copy, and redistribute my contributions, under Sentry's choice of terms.

### CODE REVIEW

**Review by viglia — APPROVED** [review]

File: src/sentry/api/endpoints/project_rules.py [hunk 1]
```diff
@@ -205,7 +205,7 @@ def _compare_lists_of_dicts(
             raw_left = {k: v for k, v in left.items() if k not in keys_to_ignore}
             raw_right = {k: v for k, v in right.items() if k not in keys_to_ignore}
 
-            # TODO (Yash): This code commented below is the corrected logic which accounts for bad key values.
+            # TODO: (Yash): This code commented below is the corrected logic which accounts for bad key values.
```
Comment (vgrozdanic) [inline src/sentry/api/endpoints/project_rules.py:208]: Here the colon char already exists, so there is no need to add it twice :)

File: src/sentry/backup/dependencies.py [hunk 2]
```diff
@@ -25,7 +25,7 @@
 #
 # NOTE TO FUTURE EDITORS: please keep the `DELETED_FIELDS` dict, and the subsequent `if` clause,
 # around even if the dict is empty, to ensure that there is a ready place to pop shims into. For
-# each entry in this dict, please leave a TODO comment pointed to a github issue for removing
+# each entry in this dict, please leave a TODO: comment pointed to a github issue for removing
```
Comment (vgrozdanic) [inline src/sentry/backup/dependencies.py:28]: TODO is mentioned in the middle of the text, no need for a colon char

File: src/sentry/backup/dependencies.py [hunk 3]
```diff
@@ -43,7 +43,7 @@
 #
 # NOTE TO FUTURE EDITORS: please keep the `DELETED_MODELS` set, and the subsequent `if` clause,
 # around even if the set is empty, to ensure that there is a ready place to pop shims into. For
-# each entry in this set, please leave a TODO comment pointed to a github issue for removing
+# each entry in this set, please leave a TODO: comment pointed to a github issue for removing
```
Comment (vgrozdanic) [inline src/sentry/backup/dependencies.py:46]: TODO is mentioned in the middle of the text, no need for a colon char

**Review by vgrozdanic — CHANGES_REQUESTED** [review]
Thank you for your contribution, we really appreciate you taking the time to make our codebase better! :)

I have left a few comments regarding the changes that you have made

File: src/sentry/api/endpoints/project_rules.py (hunk 1 above)
Comment (rettinghaus) [reply src/sentry/api/endpoints/project_rules.py:208]: That one in fact slipped me. Thanks

**Review by volokluev — APPROVED** [review]

**Review by priscilawebdev — APPROVED** [review]
thanks for improving our codebase 👏

**Review by vgrozdanic — APPROVED** [review]

**Pull request merged**